src/components/Swap/Input.tsx

```diff
@@ -52,6 +52,30 @@ export const InputColumn = styled(Column)<{ approved?: boolean; hasColor?: boole
     filter: ${({ approved }) => (approved ? undefined : 'saturate(0) opacity(0.4)')};
     transition: filter 0.25s;
   }
+
+  &:before {
+    box-sizing: border-box;
+    background-size: 100%;
+    border-radius: inherit;
+
+    position: absolute;
+    top: 0;
+    left: 0;
+
+    width: 100%;
+    height: 100%;
+    pointer-events: none;
+    content: '';
+    border: 1px solid ${({ theme }) => theme.module};
+  }
+
+  &:hover:before {
+    border-color: ${({ theme }) => theme.interactive};
+  }
+
+  &:focus-within:before {
+    border-color: ${({ theme }) => theme.outline};
```

Comment:
thoughts on using `outline` here instead of border to better align w/ standard `focus` behavior?

Reply:
similarly, why does this need to be on a pseudo-element?

Reply:
i'm guessing due to this? `Before Chrome 94, outline does not follow the shape of border-radius` https://caniuse.com/?search=outline. idk what our browser support targets are